include/meta/topics/topic_model.h

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file topic_model.h
+ * @author Matt Kelly
+ *
+ * All files in META are dual-licensed under the MIT and NCSA licenses. For more
+ * details, consult the file LICENSE.mit and LICENSE.ncsa in the root of the
+ * project.
+ */
+
+#ifndef META_TOPICS_TOPICS_H_
+#define META_TOPICS_TOPICS_H_
+
+#include <istream>
+#include <vector>
+
+#include "cpptoml.h"
+#include "meta/config.h"
+#include "meta/meta.h"
+#include "meta/stats/multinomial.h"
+#include "meta/util/aligned_allocator.h"
+#include "meta/util/fixed_heap.h"
+#include "meta/util/string_view.h"
+
+namespace meta
+{
+namespace topics
+{
+
+struct term_prob
+{
+    std::size_t tid;
+    double probability;
+};
+
+struct topic_prob
+{
+    std::size_t tid;
+    double probability;
+};
+
+/**
+ * A read-only model for accessing topic models.
+ */
+class topic_model
+{
+  public:
+    /**
+     * Load topic models from files.
+     *
+     * @param theta The stream to read the vocabulary from
+     * @param phi The stream to read the vectors from
+     */
+    topic_model(std::istream& theta, std::istream& phi);
+
+    /**
+     * @param topic_id The topic to use
+     * @param k The number of words to return
+     * @return the top k most probable words in the topic
+     */
+    std::vector<term_prob> top_k(term_id topic_id, std::size_t k = 10) const;
+
+    /**
+     * @param doc_id The document we are concerned with
+     * @return The probability of each of k topics for the
+     * given document
+     */
+    stats::multinomial<topic_id> topic_distribution(doc_id doc) const;
+
+    /**
+     * @param topic_id The topic we are concerned with
+     * @param term_id The term we are concerned with
+     * @return The probability of the term for the given topic
+     */
+    term_prob term_probability(topic_id topic_id, term_id term_id) const;
```

Comment:
Naming is hard...

I'll move it to a double and do likewise for topic_probability()